### ISSUE TITLE
[Bug] Unwanted vertical line on Material Dialog titles when launched from preferences

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#12669]
###### Reproduction Steps

1. In `DevOptionsFragment`, allow the "Enable developer options" to be shown in Debug mode (see patch)
2. Select "Enable Developer Options" to disable them. A Dialog should appear

```patch
Index: AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt b/AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt
--- a/AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt	(revision 9703ac7bb2effe4e7ec7a6a4de630df1bf0fe908)
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt	(date 1665872231649)
@@ -39,14 +39,12 @@
         val enableDevOptionsPref = requirePreference<SwitchPreference>(R.string.dev_options_enabled_by_user_key)
         // If it is a DEBUG build, hide the preference to disable developer options
         // If it is a RELEASE build, configure the preference to disable dev options
-        if (BuildConfig.DEBUG) {
-            enableDevOptionsPref.isVisible = false
-        } else {
-            enableDevOptionsPref.setOnPreferenceChangeListener { _, _ ->
-                showDisableDevOptionsDialog()
-                false
-            }
+
+        enableDevOptionsPref.setOnPreferenceChangeListener { _, _ ->
+            showDisableDevOptionsDialog()
+            false
         }
+
         // Make it possible to test crash reporting
         requirePreference<Preference>(getString(R.string.pref_trigger_crash_key)).setOnPreferenceClickListener {
             Timber.w("Crash triggered on purpose from advanced preferences in debug mode")
```

###### Expected Result

A normal Material Dialog is seen

###### Actual Result

There is a vertical line to the right of the `s` in the title's TextView, this fades after a few seconds (ignore the other lines - due to screenshotting on macOS from an emulator)

![Screenshot 2022-10-15 at 23 30 16](https://user-images.githubusercontent.com/62114487/196010026-8f20bd90-bc59-4633-abf7-330f3b9e907b.png)

###### Debug info
Tested when this file was introduced (9703ac7bb2effe4e7ec7a6a4de630df1bf0fe908) on and `main`

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison** [comment]
@BrayanDSO just in case you know anything of this (given it's preference-related).

**Comment by mikehardy** [comment]
Did this start happening after #12409 perhaps? :thinking:

**Comment by oakkitten** [comment]
Looks like a scrollbar? Anyway, can't reproduce on my device. And that screenshot is from before #12409 judging from the buttons.

**Comment by oakkitten** [comment]
But while testing I found another issue

https://user-images.githubusercontent.com/1710718/196010789-7dd43501-b68f-4d9f-817d-613b6f374e8e.mp4

**Comment by mikehardy** [comment]
> But while testing I found another issue, probably another effect of manual dealing with configuration changes.

As soon as scoped storage is done I'm wide open for any other sort of cross-cutting changes just have to maintain focus a bit longer. Subtle bug though - good catch finding that

**Comment by BrayanDSO** [comment]
I've only seen this on emulators, since months ago. IIRC, this is not exclusive to this dialog.

As I've never seen this on a real device, I've never considered this an actual issue to fix

**Comment by BrayanDSO** [comment]
> But while testing I found another issue, probably another effect of manual dealing with configuration changes.

Can't reproduce here
```
AnkiDroid Version = 2.16alpha85

Android Version = 12

Manufacturer = samsung

Model = SM-N770F

Hardware = exynos9810

Webview User Agent = Mozilla/5.0 (Linux; Android 12; SM-N770F Build/SP1A.210812.016; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/105.0.5195.136 Mobile Safari/537.36

ACRA UUID = 28fc9ae4-faa1-462f-9711-2249aec8346e

New schema = false

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

**Comment by BrayanDSO** [comment]
I'd try using an AlertDialog. If this fixes it, it's one more reason to ditch the MaterialDialog dependency

**Comment by oakkitten** [comment]
Actually, I could reproduce the scrollbar issue by changing the system setting “Display size” to “Small”.

> > But while testing I found another issue, probably another effect of manual dealing with configuration changes.

> Can't reproduce here

Try setting “Display size” to “Large” or “Larger”, I normally use “Large”. In the video when the screen is rotated the dialog stays the same width. You can see that when I close it and reopen in landscape, the width is different.

**Comment by BrayanDSO** [comment]
Here (Samsung device) I couldn't find a "Display size" setting (not even on Dev settings), but putting the font size on the biggest size showed the bars on the buttons temporarily

If it helps anyone else, `Reset languages` on "Advanced" it's an easier dialog to test. 

Funny not seeing this on other dialogs (Tested Help, Support and Restore backup). I'd still test if AlertDialog solves it.

**Comment by BrayanDSO** [comment]
Bingo. AlertDialog solves it (at least on the emulator)
